kernel/threads.c

```diff
@@ -133,6 +133,12 @@ void thread_exit (void) {
   running_tcb->state = TS_NONEXIST;
 }
 
+void thread_exit_schedule (void) {
+  thread_exit();
+  schedule();
+  panic("Rescheduled exited thread");
```

Comment:
This was already a relatively common paradigm. Note that it's *not* inlineable because it needs to be used as the exit syscall handler. I think it would also make panic'd stack traces easier to make sense of.